File: docs/src/testers/sessions/configuration/cookie_file.ts
// @sniptest filename=cookie_file.ts
// @sniptest show=1-10
import { NotteClient } from "notte-sdk";

const client = new NotteClient();

const session = client.Session({ cookie_file: "cookies.json" });
await session.use(async (session) => {
  const page = await session.page();
  await page.goto("https://example.com");
  // Cookies auto-loaded at start, auto-saved when session ends
});

const status = await session.status();
export { status };

File: docs/src/testers/sessions/configuration/cookie_file.py
# @sniptest filename=cookie_file.py
# @sniptest show=1-8
from notte_sdk import NotteClient

client = NotteClient()

with client.Session(cookie_file="cookies.json") as session:
    page = session.page
    page.goto("https://example.com")
    # Cookies auto-loaded at start, auto-saved when session ends
    cookies = session.get_cookies()
    assert any(cookie["name"] == "sniptest_cookie" and cookie["value"] == "paired" for cookie in cookies)
    status = session.status()
